src/main/java/ch/njol/skript/expressions/ExprAnvilRepairCost.java

```diff
@@ -0,0 +1,131 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.AnvilInventory;
+import org.bukkit.inventory.Inventory;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Anvil Repair Cost")
+@Description({"Returns the experience cost (in levels) to complete the current repair or the maximum experience cost (in levels) to be allowed by the current repair.",
+			  "The default value of max cost set by vanilla Minecraft is 40."})
+@Examples({
+		"on inventory click:",
+		"\tif {AnvilRepairSaleActive} = true:",
+		"\t\twait a tick # recommended, to avoid client bugs",
+		"\t\tset anvil repair cost to anvil repair cost * 50%",
+		"\t\tsend \"Anvil repair sale is ON!\" to player",
+
+		"on inventory click:",
+		"\tplayer have permission \"anvil.repair.max.bypass\"",
+		"\tset max repair cost of event-inventory to 99999"})
+@Since("INSERT VERSION")
+@RequiredPlugins("MC 1.11, MC 1.13+ (max cost)")
+public class ExprAnvilRepairCost extends SimplePropertyExpression<Inventory, Integer> {
+
+	static {
+		String maxCostSyntax = Skript.methodExists(AnvilInventory.class, "getMaximumRepairCost") ? " [:max[imum]]" : "";
+
+		if (Skript.methodExists(AnvilInventory.class, "getRepairCost"))
+			Skript.registerExpression(ExprAnvilRepairCost.class, Integer.class, ExpressionType.PROPERTY,
+				"[the] [anvil] [item]" + maxCostSyntax + " repair cost [of %inventories%]",
+				"%inventories%'[s] [item]" + maxCostSyntax + " repair cost");
+	}
+
+	boolean isMax = false;
+	Expression<Inventory> invs;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		invs = (Expression<Inventory>) exprs[0];
```

Comment:
since you are using setExpr this is not needed

you can use `getExpr()` in the other methods :)